test/insertion_order_parser_test.dart

```diff
@@ -0,0 +1,375 @@
+import 'package:angular_test/angular_test.dart';
+import 'package:test/test.dart';
+
+import 'package:dv360_excel_plugin/src/insertion_order_parser.dart';
+import 'package:dv360_excel_plugin/src/proto/insertion_order.pb.dart';
+
+void main() {
+  String input;
+  const emptyEntry = '';
+
+  InsertionOrder insertionOrderTemplate;
+  InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange
+      oneDateRange;
+  InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange_Date
+      oneStartDate;
+  InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange_Date
+      oneEndDate;
+
+  setUp(() {
+    insertionOrderTemplate = InsertionOrder()
+      ..advertiserId = emptyEntry
+      ..campaignId = emptyEntry
+      ..insertionOrderId = emptyEntry
+      ..displayName = emptyEntry
+      ..updateTime = emptyEntry
+      ..entityStatus = InsertionOrder_EntityStatus.ENTITY_STATUS_UNSPECIFIED
+      ..pacing = InsertionOrder_Pacing()
+      ..budget = InsertionOrder_InsertionOrderBudget();
+
+    oneStartDate =
+        InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange_Date()
+          ..year = 2019
+          ..month = 1
+          ..day = 1;
+
+    oneEndDate =
+        InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange_Date()
+          ..year = 2019
+          ..month = 12
+          ..day = 31;
+
+    oneDateRange =
+        InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange()
+          ..startDate = oneStartDate
+          ..endDate = oneEndDate;
+  });
+
+  tearDown(disposeAnyRunningTest);
+
+  group('parse insertionOrder from empty json:', () {
+    setUp(() => input = '{}');
+
+    test('function should return normally', () {
```

Comment:
can delete this test since if it throws "result is an empty insertionOrder instance" will catch it. We by default assume a method should return normally. We do encourage testing when that isnt the case though, e.g. we expect it to throw